src/eko/io/inventory.py

```diff
@@ -102,10 +102,9 @@ def __getitem__(self, header: H) -> Optional[Operator]:
 
         # in case of contentless, check header availability instead
         if self.contentless:
-            if self.lookup(stem, header=True):
-                self.cache[header] = None
-                return None
-            raise LookupError(f"Impossible to retrieve {header} from {self}.")
+            self.lookup(stem, header=True)
+            self.cache[header] = None
+            return None
```

Comment:
I would revert this, since they are two different situations:
1. you request a `header` that is present, but not loaded in `cache`, thus you put in the `cache`
2. you request a `header` that is not present, consequently you should receive an error

With your change, you can add elements to the cache by requesting them, and they would never be associated to items on disk, so you would lose them after the run.

Reply:
Ah, you're right. I just forgot `lookup` was raising :P